bark/models/behavior/safety_behavior/safety_behavior.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#ifndef BARK_MODELS_BEHAVIOR_SAFETY_BEHAVIOR_SAFETY_BEHAVIOR_HPP_
+#define BARK_MODELS_BEHAVIOR_SAFETY_BEHAVIOR_SAFETY_BEHAVIOR_HPP_
+
+#include <memory>
+#include <utility>
+
+#include "bark/models/behavior/behavior_model.hpp"
+#include "bark/models/behavior/idm/idm_lane_tracking.hpp"
+#include "bark/world/world.hpp"
+
+namespace bark {
+namespace models {
+namespace behavior {
+
+using dynamic::Trajectory;
+using world::ObservedWorld;
+using world::objects::AgentId;
+using bark::models::behavior::BehaviorIDMLaneTracking;
+using bark::world::map::LaneCorridor;
+using bark::world::map::LaneCorridorPtr;
+using bark::commons::ParamsPtr;
+
+class BehaviorSafety : public BehaviorModel {
+ public:
+  explicit BehaviorSafety(const ParamsPtr& params) :
+    BehaviorModel(params),
+    initial_lane_corr_(nullptr) {
+      safety_behavior_params_ = params->AddChild("BehaviorSafety");
+      safety_behavior_params_->SetReal("BehaviorIDMClassic::DesiredVelocity", 1.);
+      // safety_behavior_params_->SetReal(
+      //   "BehaviorIDMClassic::AccelerationLowerBound", -0.5);
+      behavior_model_ = std::make_shared<BehaviorIDMLaneTracking>(
+        safety_behavior_params_);
+  }
+
+  virtual ~BehaviorSafety() {}
+
+  Trajectory Plan(float min_planning_time, const ObservedWorld& observed_world);
+
+  virtual std::shared_ptr<BehaviorModel> Clone() const;
+
+  void SetBehaviorModel(const std::shared_ptr<BehaviorModel>& model){
+    behavior_model_ = model;
+  } 
+
+  void SetInitialLaneCorridor(const LaneCorridorPtr& lc){
+    initial_lane_corr_ = lc;
+  }
+
+  ParamsPtr GetBehaviorSafetyParams() const {
+    return safety_behavior_params_;
+  }
+
+ private:
+  std::shared_ptr<BehaviorModel> behavior_model_;
```

Comment:
better naming required for this member. Change to specific IDM type or derive from this type.

Reply:
would leave it generic see comment below